shell/core/types.ts

```diff
@@ -17,7 +17,17 @@ export interface PackageMetadata {
 
 export type VuexStoreObject = { [key: string]: any }
```

Comment:
We may create `.d.ts` distributed files so the there's no need to user export and then import the types.

Reply:
It will require to adjust all the linting configuration though, as I'd to do as well. There's an issue about it already.
https://github.com/rancher/dashboard/issues/6537